### ISSUE TITLE
logging of path, method and remote host from axum

### ISSUE THREAD

**Issue by yaleman** [kanidm/kanidm#1820]
**Is your feature request related to a problem? Please describe.**

These are missing from INFO level logs

**Describe the solution you'd like**

Logs

**Describe alternatives you've considered**

Not logs

**Additional context**

**Comment by yaleman** [comment]
`TrustedClientIP` should really be on *all* requests so we can log remote activity, if it's implemented as an `Extension` then it can be queried anywhere in the request flow.

**Comment by Firstyear** [comment]
We used to log this out of the same place as we did with kopid now, and because it's an extractor we can actually use it inside kopid still even with the extractor pattern. I think my only worry would be double-extracting on some requests. But even then, that's a minor thing imo. Later maybe we'll need this for IP bound tokens so we'll probably need it near by the uat anyway.

**Comment by yaleman** [comment]
#1960 will largely solve this, but not for the `Forwarded` IPs, but that can get sorted.

**Comment by NavinShrinivas** [comment]
Hey @yaleman why is this tagged blocked? Also, the merged PR seems to solve almost everything...does this still require some cleanup? As for forwarded IPs, does Axum get details about 302's?

**Comment by yaleman** [comment]
Yeah, it's not blocked anymore, it was ... based on something else at the time. We should be logging the `Forwarded` IP if we're trusting it (or the legacy `x-forwarded-for` header) - I believe we're still only logging the connecting IP.

302's have nothing to do with Forwarded IPs, I'm not sure what you mean by that one?

**Comment by Samonitari** [comment]
I have a hard time narrowing down the issue with caddy and proxy-v2.
Enabling it on both caddy side and kanidm side gets me `Failed to create TLS stream`, but disabling it only on caddy side works, even though I have no `http_client_address_info.x-forward-for`  array just `proxy-v2`.
Long story short, I would've loved to exclude potential issues with the client IP, seems like I can't - yet - inspect this.

Thanks for all the hard work and love you've put it kanidm, it's just a roundabout request with the same reason that the reporter of #3609 gave...

**Comment by yaleman** [comment]
@Samonitari it should be logging `connection_addr` and `client_ip_addr` or throwing an error if it can't parse the data out - what version are you running?

**Comment by Firstyear** [comment]
Given that this issue is about logging, checking my logs I can confirm we now log path method and remote host ip's correctly so we can close this. 

@Samonitari As you have a support question, please open a discussion instead.
